apps/dashboard/src/components/billing/highlights-row.tsx

```diff
@@ -1,33 +1,23 @@
 import { Badge } from '@/components/primitives/badge';
 import { Card } from '@/components/primitives/card';
-import { ApiServiceLevelEnum } from '@novu/shared';
+import { ApiServiceLevelEnum, FeatureNameEnum, getFeatureForTierAsNumber, getFeatureForTierAsText } from '@novu/shared';
 
-interface Highlight {
-  text: string;
-  badgeLabel?: string;
-}
-
-type PlanHighlights = {
-  [key in ApiServiceLevelEnum]?: Highlight[];
+const serviceLevelToFuncArray: Record<string, ((ApiServiceLevelEnum: ApiServiceLevelEnum) => string)[]> = {
+  [ApiServiceLevelEnum.FREE_2025]: [getEventsLine, getTeammatesLine, feedRetentionLine],
+  [ApiServiceLevelEnum.PRO]: [getEventsLine, getTeammatesLine, feedRetentionLine],
+  [ApiServiceLevelEnum.TEAM]: [getEventsLine, getTeammatesLine, feedRetentionLine],
+  [ApiServiceLevelEnum.ENTERPRISE]: [getEventsLine, getTeammatesLine, getSamlText],
 };
 
-const highlights: PlanHighlights = {
-  [ApiServiceLevelEnum.FREE]: [
-    { text: 'Up to 30,000 events per month' },
-    { text: '3 teammates' },
-    { text: '30 days Activity Feed retention' },
-  ],
-  [ApiServiceLevelEnum.BUSINESS]: [
-    { text: 'Up to 250,000 events per month' },
-    { text: 'Unlimited teammates' },
-    { text: '90 days Activity Feed retention' },
-  ],
-  [ApiServiceLevelEnum.ENTERPRISE]: [
-    { text: 'Up to 5,000,000 events per month' },
-    { text: 'Unlimited teammates' },
-    { text: 'SAML SSO' },
-  ],
-};
+const highlights: PlanHighlights = Object.entries(serviceLevelToFuncArray).reduce(
+  (highlights, [serviceLevelKey, funcs]) => {
+    highlights[serviceLevelKey as ApiServiceLevelEnum] = funcs.map((func) => ({
+      text: func(serviceLevelKey as ApiServiceLevelEnum),
+    }));
+    return highlights;
+  },
+  {} as PlanHighlights
+);
```

Comment:
Same as before, I would keep it hardcoded as it was - this requires significant mental overhead to understand